tests_e2e/tests/ext_policy/ext_policy.py

```diff
@@ -0,0 +1,233 @@
+        )
+        unique2 = str(uuid.uuid4())
+        test_file2 = f"waagent-test.{unique2}"
+        run_command_2 = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.RunCommandHandler,
+                                          resource_name="RunCommandHandler2"),
+            {'source': {'script': f"echo '{unique2}' > /tmp/{test_file2}"}}
+        )
+
+        # AzureMonitorLinuxAgent is a no-config extension (extension without settings).
+        azure_monitor = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.AzureMonitorLinuxAgent,
+                                          resource_name="AzureMonitorLinuxAgent"),
+            None
+        )
+
+        # Enable policy via conf
+        log.info("Enabling policy via conf file on the test VM [%s]", self._context.vm.name)
+        self._ssh_client.run_command("update-waagent-conf Debug.EnableExtensionPolicy=y", use_sudo=True)
+
+        # Only allowlisted extensions should be processed.
+        # We only allowlist CustomScript: CustomScript should be enabled, RunCommand and AzureMonitor should fail.
+        # (Note that CustomScript blocked by policy is tested in a later test case.)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {
+                        "Microsoft.Azure.Extensions.CustomScript": {}
+                    }
+                }
+            }
+        self._create_policy_file(policy)
+        self._operation_should_succeed("enable", custom_script)
+        self._operation_should_fail("enable", run_command)
+        if VmExtensionIds.AzureMonitorLinuxAgent.supports_distro((self._ssh_client.run_command("get_distro.py").rstrip())):
+            self._operation_should_fail("enable", azure_monitor)
+
+        # When allowlist is turned off, all extensions should be processed.
+        # RunCommand and AzureMonitorLinuxAgent should be successfully enabled and then deleted.
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": False,
+                    "signatureRequired": False,
+                    "extensions": {}
+                }
+            }
+        self._create_policy_file(policy)
+        self._operation_should_succeed("enable", run_command)
+        self._operation_should_succeed("delete", run_command)
+        if VmExtensionIds.AzureMonitorLinuxAgent.supports_distro((self._ssh_client.run_command("get_distro.py").rstrip())):
+            self._operation_should_succeed("enable", azure_monitor)
+            self._operation_should_succeed("delete", azure_monitor)
+
+        # Should not uninstall disallowed extensions.
```

Comment:
seems like

    # Only allowlisted extensions should be processed.

and

     # Should not uninstall disallowed extensions.

or are we trying to test something different?

Reply:
I've updated the comments to make it more clear what is being tested with each policy